### ISSUE TITLE
User Warning implicit cast from 'char *' to a different pointer type

### ISSUE THREAD

**Issue by jborean93** [pyca/pyopenssl#568]
Hi in one of my packages that calls PyOpenSSL it seems like I get the below warning everytime it runs

```
/apps/dev/venvs/python27/lib/python2.7/site-packages/OpenSSL/crypto.py:1108: UserWarning: implicit cast from 'char *' to a different pointer type: will be forbidden in the future (check that the types are as you expect; use an explicit ffi.cast() if they are correct)
  self._x509, digest, result_buffer, result_length)
```

Here is my installed libraries through pip
```
ansible (2.2.0.0)
ansible-lint (3.3.0)
ansible-review (0.10.1)
appdirs (1.4.0)
awscli (1.10.66)
boto (2.42.0)
botocore (1.4.56)
cffi (1.8.3)
cloud-accounts-client (1.11)
colorama (0.3.7)
configparser (3.5.0)
cryptography (1.5.2)
docutils (0.12)
enum34 (1.1.6)
flake8 (3.0.4)
future (0.15.2)
futures (3.0.5)
idna (2.1)
ipaddress (1.0.17)
Jinja2 (2.8)
jmespath (0.9.0)
MarkupSafe (0.23)
mccabe (0.5.2)
ntlm-auth (1.0.2)
ordereddict (1.1)
paramiko (2.0.2)
pip (8.1.2)
pyasn1 (0.1.9)
pycodestyle (2.0.0)
pycparser (2.17)
pycrypto (2.6.1)
pyflakes (1.2.3)
pyOpenSSL (16.1.0)
python-dateutil (2.5.3)
python-ntlm3 (1.0.2)
pywinrm (0.2.1)
PyYAML (3.12)
requests (2.11.1)
requests-credssp (0.0.1)
requests-ntlm (0.4.0)
rsa (3.4.2)
s3transfer (0.1.4)
setuptools (28.7.1)
six (1.10.0)
unidiff (0.5.2)
wheel (0.29.0)
xmltodict (0.10.2)
```

Looking at https://github.com/pyca/cryptography/issues/3167 this seems to have been fixed in newer version but this doesn't seem to be the case. I am not sure where the issue lies and if this should be an issue raised in another library.

**Comment by reaperhulk** [comment]
I believe this was fixed in pyOpenSSL 16.2 (you have 16.1)

**Comment by jborean93** [comment]
Thanks for the info I should have tried that first.
